### PR TITLE
fix(lottie.ts): passing credentials include param to fetch request

### PR DESCRIPTION
**Pull request by akil3**
Fetch request is failing with 403 when trying to fetch lottie json hosted on another domain that gates the assests behind credentials. https://javascript.info/fetch-crossorigin#credentials

### CODE REVIEW

**Review by mrloop — CHANGES_REQUESTED** [review]
Thanks for the PR. I don't think this should be the default, but configurable. How about a named argument `@fetchOptions` so the call `await fetch(this.args.path, this.args.fetchOptions)` can be made? Would need a test, maybe `sinon` can be used, its already a dependency for the tests, to mock `fetch` and check options are passed correctly?

**Comment by akil3** [comment]
@mrloop added `fetchOptions` argument as suggested and couple test cases for the same.

File: test-app/tests/integration/components/lottie-test.ts [hunk 1]
```diff
@@ -16,11 +13,12 @@ interface TestContext extends TestContextBase {
 
 module('Integration | Component | lottie', function (hooks) {
   setupRenderingTest(hooks);
-  setupWindowMock(hooks);
```
Comment (akil3) [inline test-app/tests/integration/components/lottie-test.ts:19]: All tests are passing without this. 
Moreover `fetch` is not present in the `window` mock in order to spy on.

**Review by mrloop — CHANGES_REQUESTED** [review]
Thanks for the update, left a couple of comments about removing `ember-window-mock`

File: test-app/tests/integration/components/lottie-test.ts (hunk 1 above)
Comment (akil3) [reply test-app/tests/integration/components/lottie-test.ts]: @mrloop sinon spying and stubbing will work now. 
There was an issue with latest sinon and ember-window-mock proxy implementation. It has been fixed in the latest version here: https://github.com/simonihmig/ember-window-mock/pull/535

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -1,4 +1,8 @@
-lockfileVersion: '6.0'
+lockfileVersion: '6.1'
```
Comment (mrloop) [inline pnpm-lock.yaml:1]: Looks like your using pnpm v8.6.1, there was breaking change with this release which was fixed in v8.6.2. See https://github.com/pnpm/pnpm/issues/6648. If you upgrade pnpm and then upgrade ember-window-mock again then the lock file wont change it's lock file version. We want to remain on '6.0'. Everything else looks good

File: test-app/tests/integration/components/lottie-test.ts (hunk 1 above)
Comment (mrloop) [reply test-app/tests/integration/components/lottie-test.ts]: Great, thank you!

File: pnpm-lock.yaml (hunk 2 above)
Comment (akil3) [reply pnpm-lock.yaml]: Upgraded pnpm to v8.6.2 and installed the updated the package

Comment (mrloop) [reply pnpm-lock.yaml]: OK, I'd remove those settings. Also the build is failing for unrelated issues, it should pass if you rebase on main, I spent a bit of time yesterday merging outstanding PRs which fixed the failing build. Ideally this PR would be in two commits the first commit with the dependency update and the second with the feature update. Alternatively one commit would be fine, with all of them squashed. This means we'll have a clean history, and one which all commit should build and pass test suite, helpful if you ever need to git bisect the repo.

Comment (mrloop) [reply pnpm-lock.yaml]: Quick update, do want merge this PR, but ember-window-mock is a breaking change due to it dropping support for ember 4.4. So in another PR me or another maintainer will drop 4.4 and other unsupported ember versions. 4.4 last security updates are in a months time. The PR is also failing on other ember version but think its a CI issue, so I need to fix the build pipeline. Not sure of the timeline

Comment (akil3) [reply pnpm-lock.yaml]: Thanks for the update @mrloop

**Comment by nicolasgasco** [comment]
Hi, @akil3! Thank you for your interest in `ember-lottie`. What's the outcome of your discussion with @mrloop ? Do you still mean to work on this PR or would you rather close it for now?

**Comment by mrloop** [comment]
I've rebased and cleaned up in https://github.com/qonto/ember-lottie/pull/274, closing this MR in favour of that one

**Pull request closed without merging**